aviatrix/utils.go

```diff
@@ -103,3 +104,29 @@ func intInSlice(needle int, haystack []int) bool {
 	}
 	return false
 }
+
+func extractTags(d *schema.ResourceData, cloudType int) (map[string]string, error) {
```

Comment:
Nitpick: It is more idiomatic Go style to keep the "happy path" (non-error case) aligned on the left, and leave the indented parts to the error handling. For this function it would look like
```golang
func extractTags(d *schema.ResourceData, cloudType int) (map[string]string, error) {
	tags, ok := d.GetOk("tags")
	 if ok && !intInSlice(cloudType, []int{goaviatrix.AWS, goaviatrix.AWSGOV,goaviatrix.AZURE} {
		return nil, fmt.Errorf("adding tags is only supported for AWS, AWSGOV and AZURE, cloud_type must be 1, 256 or 8")
	}
	 if !ok {
	 	 return nil, nil
	 }
	tagsMap := tags.(map[string]interface{})
	tagsStrMap := make(map[string]string, len(tagsMap))
	for key, val := range tagsMap {
		valStr := fmt.Sprint(val)
		if strings.Contains(key, ",") {
			return nil, fmt.Errorf("illegal character in tags: \",\"")
		}
		matched, err := regexp.MatchString("^[a-zA-Z0-9+\\-=._ :/@ ]*$", key+valStr)
		if err != nil || !matched {
			return nil, fmt.Errorf("illegal characters in tags")
		}
		escapedKey := strings.ReplaceAll(key, ":", "\\:")
		escapedVal := strings.ReplaceAll(valStr, ":", "\\:")
		tagsStrMap[escapedKey] = escapedVal
	}
	return tagsStrMap, nil
}
```

We end up with a less indented function that is easier to read.